src/Survey/ItechArt.Survey.Foundation/Validation/UserValidator.cs

```diff
@@ -0,0 +1,109 @@
+    {
+        var validationResult = ValidateUserName(user.UserName, options);
+
+        if (validationResult.HasError)
+        {
+            return ValidationResult<UserRegistrationErrors>.CreateResultWithError(validationResult.Error);
+        }
+
+        validationResult = ValidateEmail(user.Email, options);
+
+        if (validationResult.HasError)
+        {
+            return ValidationResult<UserRegistrationErrors>.CreateResultWithError(validationResult.Error);
+        }
+
+        return ValidationResult<UserRegistrationErrors>.CreateResultWithoutError();
+    }
+
+    public ValidationResult<UserRegistrationErrors> Validate(User user, string password, RegistrationOptions options)
+    {
+        var validationResult = Validate(user, options);
+
+        if (validationResult.HasError)
+        {
+            return ValidationResult<UserRegistrationErrors>.CreateResultWithError(validationResult.Error);
+        }
+
+        validationResult = ValidatePassword(password, options);
+
+        if (validationResult.HasError)
+        {
+            return ValidationResult<UserRegistrationErrors>.CreateResultWithError(validationResult.Error);
+        }
+
+        return ValidationResult<UserRegistrationErrors>.CreateResultWithoutError();
+    }
+
+
+    private ValidationResult<UserRegistrationErrors> ValidateUserName(string userName, RegistrationOptions options)
+    {
+        if (string.IsNullOrEmpty(userName))
+        {
+            return ValidationResult<UserRegistrationErrors>.CreateResultWithError(UserRegistrationErrors.UserNameIsRequired);
+        }
+
+        if (!(options.UserNameMinLength <= userName.Length && userName.Length <= options.UserNameMaxLength))
+        {
+            return ValidationResult<UserRegistrationErrors>.CreateResultWithError(UserRegistrationErrors.InvalidUserNameLength);
+        }
+
+        var match = options.UserNamePattern.Match(userName);
+
+        if (string.IsNullOrEmpty(match.Value))
+        {
+            return ValidationResult<UserRegistrationErrors>.CreateResultWithError(UserRegistrationErrors.IncorrectUserName);
+        }
+
+        return ValidationResult<UserRegistrationErrors>.CreateResultWithoutError();
+    }
+
+    private ValidationResult<UserRegistrationErrors> ValidateEmail(string email, RegistrationOptions options)
+    {
+        if (string.IsNullOrEmpty(email))
+        {
+            return ValidationResult<UserRegistrationErrors>.CreateResultWithError(UserRegistrationErrors.EmailIsRequired);
+        }
+
+        var match = options.EmailPattern.Match(email);
+
+        if (string.IsNullOrEmpty(match.Value))
+        {
+            return ValidationResult<UserRegistrationErrors>.CreateResultWithError(UserRegistrationErrors.IncorrectEmail);
+        }
+
+        return ValidationResult<UserRegistrationErrors>.CreateResultWithoutError();
+    }
+
+    private ValidationResult<UserRegistrationErrors> ValidatePassword(string password, RegistrationOptions options)
```

Comment:
Possible solutions:
1. Move password validation to authentication service
2. ValidationResult<PasswordValidationError> IPasswordValidator.Validate(string password)
3. Take a look at how ASP.NET Core Identity validates password (see sources on GitHub, etc.)